File: src/Java/IO/DEMReader.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkDEMReader;
import vtk.vtkLookupTable;
import vtk.vtkImageMapToColors;
import vtk.vtkImageActor;
import vtk.vtkInteractorStyleImage;

public class DEMReader 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {

    //parse command line arguments
    if (args.length != 1) 
    {
      System.err.println("Usage: java -classpath ... Filename(.dem) e.g SainteHelens.dem");
      return;
    }
    String inputFilename = args[0];
    
    //Read the file
    vtkDEMReader reader = new vtkDEMReader();
    reader.SetFileName(inputFilename);
    reader.Update();
    
    vtkLookupTable lut = new vtkLookupTable();
    lut.SetHueRange(0.6, 0);
    lut.SetSaturationRange(1.0, 0);
    lut.SetValueRange(0.5, 1.0);
    lut.SetTableRange(reader.GetOutput().GetScalarRange());
    
    //Visualize
    vtkImageMapToColors mapColors = new vtkImageMapToColors();
    mapColors.SetLookupTable(lut);
    mapColors.SetInputConnection(reader.GetOutputPort());
    
    //Create an actor
    vtkImageActor actor = new vtkImageActor();
    actor.GetMapper().SetInputConnection(mapColors.GetOutputPort());
    
    //Setup renderer
    vtkRenderer renderer = new vtkRenderer();
    renderer.AddActor(actor);
    renderer.ResetCamera();
    
    //Setup render window
    vtkRenderWindow renderWindow = new vtkRenderWindow();
    renderWindow.AddRenderer(renderer);
    
    //Setup render window interactor
    vtkRenderWindowInteractor renderWindowInteractor = new vtkRenderWindowInteractor();
    vtkInteractorStyleImage style = new vtkInteractorStyleImage();
    
    renderWindowInteractor.SetInteractorStyle(style);
    
    //Render and start interaction
    renderWindowInteractor.SetRenderWindow(renderWindow);
    renderWindow.Render();

    renderWindowInteractor.Initialize();
    renderWindowInteractor.Start();

  }
}

File: src/Cxx/IO/DEMReader.cxx
#include <vtkSmartPointer.h>
#include <vtkImageActor.h>
#include <vtkImageMapToColors.h>
#include <vtkLookupTable.h>
#include <vtkDEMReader.h>
#include <vtkImageData.h>
#include <vtkImageMapper3D.h>
#include <vtkRenderWindow.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkInteractorStyleImage.h>
#include <vtkRenderer.h>

int main(int argc, char* argv[])
{
  // Verify arguments
  if(argc < 2)
  {
    std::cerr << "Required: filename.dem e.g. SainteHelens.dem" << std::endl;
    return EXIT_FAILURE;
  }

  // Read the file
  vtkSmartPointer<vtkDEMReader> reader =
    vtkSmartPointer<vtkDEMReader>::New();
  reader->SetFileName(argv[1]);
  reader->Update();

  vtkSmartPointer<vtkLookupTable> lut =
    vtkSmartPointer<vtkLookupTable>::New();
  lut->SetHueRange(0.6, 0);
  lut->SetSaturationRange(1.0, 0);
  lut->SetValueRange(0.5, 1.0);
  lut->SetTableRange(reader->GetOutput()->GetScalarRange());

  // Visualize
  vtkSmartPointer<vtkImageMapToColors> mapColors =
    vtkSmartPointer<vtkImageMapToColors>::New();
  mapColors->SetLookupTable(lut);
  mapColors->SetInputConnection(reader->GetOutputPort());

  // Create an actor
  vtkSmartPointer<vtkImageActor> actor =
    vtkSmartPointer<vtkImageActor>::New();
  actor->GetMapper()->SetInputConnection(mapColors->GetOutputPort());

  // Setup renderer
  vtkSmartPointer<vtkRenderer> renderer =
    vtkSmartPointer<vtkRenderer>::New();
  renderer->AddActor(actor);
  renderer->ResetCamera();

  // Setup render window
  vtkSmartPointer<vtkRenderWindow> renderWindow =
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);

  // Setup render window interactor
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor =
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  vtkSmartPointer<vtkInteractorStyleImage> style =
    vtkSmartPointer<vtkInteractorStyleImage>::New();

  renderWindowInteractor->SetInteractorStyle(style);

  // Render and start interaction
  renderWindowInteractor->SetRenderWindow(renderWindow);
  renderWindow->Render();
  renderWindowInteractor->Initialize();

  renderWindowInteractor->Start();

  return EXIT_SUCCESS;
}
